ftw/keywordwidget/resources/js/keywordwidget.js

```diff
@@ -1,9 +1,49 @@
-$(function() {
+window.ftwKeywordWidget = function($) {
 
-    function initSelect2(widget){
+    "use strict";
+
+    var self = {};
+    var templates = {};
+
+    var init = function() {
+
+        // Special template to indicate new terms
+        registerTemplate("defaultResultTemplate", function (data) {
+            if (!data.loading && !data._resultId) {
+                return $('<span class="newTag" />')
+                .text(data.text)
+                .append($('<span class="newTagHint" />').text(i18n.label_new));
+            } else {
+                return data.text;
+            }
+        });
+
+        document.dispatchEvent(new Event('ftwKeywordWidgetInit'));
+    };
+
+    var registerTemplate = function(name, templateFunction) {
+        if (templates.hasOwnProperty(name)) {
+            console.warn("A template with the name '" + name + "' is alredy registred.");
+            return;
+        }
+        templates[name] = templateFunction;
```

Comment:
Here I would make a check if the `templateFunction` is a function -> https://api.jquery.com/jQuery.isFunction/